api/api/serializers/image_serializers.py

```diff
@@ -119,21 +119,33 @@ class Meta:
 class OembedRequestSerializer(serializers.Serializer):
     """Parse and validate oEmbed parameters."""
 
-    url = serializers.CharField(
+    url = serializers.URLField(
+        allow_blank=False,
         help_text="The link to an image present in Openverse.",
     )
 
-    @staticmethod
-    def validate_url(value):
-        url = add_protocol(value)
+    def to_internal_value(self, data):
+        data = super().to_internal_value(data)
+
+        url = data["url"]
         if url.endswith("/"):
             url = url[:-1]
         identifier = url.rsplit("/", 1)[1]
+
         try:
             uuid = UUID(identifier)
+            image = Image.objects.get(identifier=uuid)
         except ValueError:
-            raise serializers.ValidationError("Could not parse identifier from URL.")
-        return uuid
+            raise serializers.ValidationError(
+                {"Could not parse identifier from URL.": data["url"]}
+            )
+        except (Image.DoesNotExist, ValidationError):
+            raise serializers.ValidationError(
+                {"Could not find image from the provided URL": data["url"]}
+            )
```

Comment:
Why not use `clean_url` as the function to do this validation? It seems like what were doing here is field-level validation.

Reply:
@sarayourfriend, the thing here is that @AetherUnbound always suggests combining the try/excepts for slightly better performance 😄 but I agree it'd be clearer here if it's separated.

@dhruvkb What `clean_url` function are you referencing here? Or are you saying to create a new one specifically for that?

Reply:
Interesting. I've heard this before, but [try/except are very fast, according to Python documentation](https://docs.python.org/3/faq/design.html#how-fast-are-exceptions) and specifically only catching the exception is expensive. Considering catching the exception only happens once here and always results in a similar end (validation error) there isn't much of a difference I think.

Based on [this SO error](https://stackoverflow.com/questions/2522005/cost-of-exception-handlers-in-python) I could see it making sense if inside a specific tight loop as a last-ditch effort at optimisation. But in a situation like this and in probably 99.99% of our code, it makes the code significantly worse to read with literally no performance difference (because only one exception is ever handled) or otherwise with at least no _meaningful_ performance difference.

Reply:
@sarayourfriend Agree! I'm not very much concerned with performance related to these instructions either. Change applied ✅ 

@dhruvkb Now I remember. To me, it seemed more accurate to use the `to_internal_value` function because we're not only returning the URL here but also the image info, look at line 149.

Thanks for reviewing folks!

Reply:
I already approved it because I don't have any strong preference towards `validate_<field>`. Your explanation makes sense @krysal, thank you.